tests/unit/inference/test_inference.py

```diff
@@ -252,14 +266,28 @@ def test(
         torch.cuda.synchronize()
         bs_time = time.time() - start
 
-        pipe.model = deepspeed.init_inference(
-            pipe.model,
-            mp_size=1,
-            dtype=dtype,
-            replace_method="auto",
-            replace_with_kernel_inject=True,
-            enable_cuda_graph=enable_cuda_graph,
-        )
+        if "t5-v1_1-small" in model:
+            pipe.model = deepspeed.init_inference(
+                pipe.model,
+                mp_size=1,
+                dtype=dtype,
+                injection_policy={
+                    T5Block: ('SelfAttention.o',
+                              'EncDecAttention.o',
+                              'DenseReluDense.wo')
+                },
+                enable_cuda_graph=enable_cuda_graph,
+            )
```

Comment:
Adding additional if statements to the main test function could become messy. I think this is ok for now, but we should think about how we can have a single `init_inference` call.

Reply:
Thanks for the suggestion. After our discussion, we decided to refactor the testing to include a `TestInjectionPolicy` class, specifically designed to test explicit injection policies.